### PR TITLE
Set train mode after evaluation in training

### LINKED ISSUE #310: Switch between model.train() and model.eval()
https://github.com/asyml/texar-pytorch/blob/95f3d644680656f38b7b1779f2ba184ef9e1f81a/examples/bert/bert_classifier_main.py#L164

Hello, 
     I find that after doing "_eval_epoch()"，you don't switch model.eval() to model.train(). Wouldn't that be a problem?

### PR DESCRIPTION
**Pull request by AvinashBukkittu**
This PR 
- Fixes a bug in BERT and GPT2 examples of not setting mode to train after evaluation

Fixes #310

### CODE REVIEW

**Comment by huzecong** [comment]
Have you reran the fine-tuning experiments? It would be best to ensure that we're doing it right, and that the results improve compared to when we did it wrong. We might also want to update the README with the new outputs.

**Comment by AvinashBukkittu** [comment]
> Have you reran the fine-tuning experiments? It would be best to ensure that we're doing it right, and that the results improve compared to when we did it wrong. We might also want to update the README with the new outputs.

I did not completely re-run the experiments. I re-ran until the first `_eval_epoch` call in training, checked the value of mode and current dataset in the iterator right after this call (the mode was `eval` and current dataset was eval dataset, which is wrong). 

Sure, I can update the Readme with the new results.

**Comment by huzecong** [comment]
Hmm, interesting... The loss values now are much (one order of magnitude) lower than before starting from iter 250, and it seems a bit weird to me. Also, I checked the code at the commit where the README was modified to include the results:
https://github.com/asyml/texar-pytorch/blob/e4b68188388dbaa07791528d066b410a6f838de7/examples/bert/config_data.py#L9

It seems that `eval_step` was set to -1, which means no evaluation is performed until the end of training. In this case, the previous results were not incorrect (although the implementation was still faulty).

**Comment by AvinashBukkittu** [comment]
> Hmm, interesting... The loss values now are much (one order of magnitude) lower than before starting from iter 250, and it seems a bit weird to me. Also, I checked the code at the commit where the README was modified to include the results:
> https://github.com/asyml/texar-pytorch/blob/e4b68188388dbaa07791528d066b410a6f838de7/examples/bert/config_data.py#L9
> 
> It seems that `eval_step` was set to -1, which means no evaluation is performed until the end of training. In this case, the previous results were not incorrect (although the implementation was still faulty).

We start noticing a stark difference for 300th iteration. Agree, previously, after first call `_eval_epoch()`, the mode was still set to `eval`
